### PR TITLE
[Feature]add multi-task learning module and a method "PCGrad"

### PR DESCRIPTION
**Pull request by HydrogenSulfate**
### PR types

New features
### PR changes

APIs 
### Describe

主要修改点：
- [x] 添加多任务学习模块 `ppsci.loss.mtl`
- [x] 添加多任务学习算法 `PCGrad`(https://papers.nips.cc/paper/2020/hash/3fe78a8acf5fda99de95303940a2420c-Abstract.html, NIPS 2020)
- [x] 添加多任务学习模块启动方式

### CODE REVIEW

File: ppsci/solver/train.py [hunk 1]
```diff
@@ -94,9 +94,19 @@ def train_epoch_func(solver: "solver.Solver", epoch_id: int, log_freq: int):
...
+                        constraint_losses, solver.global_step
+                    ).backward()
             else:
-                total_loss.backward()
+                if solver.loss_aggregator is None:
+                    total_loss.backward()
+                else:
+                    solver.loss_aggregator(
+                        constraint_losses, solver.global_step
+                    ).backward()
```
Comment (zhiminzhang0830) [inline ppsci/solver/train.py:109]: 这块可否改成如下的形式，这样感觉会简洁一些 
```
if solver.loss_aggregator is None:
     if solver.use_amp:
        total_loss_scaled = solver.scaler.scale(total_loss)
        total_loss_scaled.backward()
    else:
        total_loss.backward()
else:
    solver.loss_aggregator(constraint_losses, solver.global_step).backward()
 ```

File: ppsci/loss/mtl/__init__.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "LossAggregator",
+    "PCGrad",
+]
+
+
+def build_mtl_aggregator(cfg: Dict[str, Any]):
+    """Build loss aggregator with multi-task learning method.
+
+    Args:
+        cfg (AttrDict): Aggregator config.
```
Comment (zhiminzhang0830) [inline ppsci/loss/mtl/__init__.py:32]: 这里是AttrDict 还是 Dict[str, Any]

File: ppsci/solver/train.py (hunk 1 above)
Comment (HydrogenSulfate) [reply ppsci/solver/train.py]: 确实这样更简洁，已修改

File: ppsci/loss/mtl/__init__.py (hunk 2 above)
Comment (HydrogenSulfate) [reply ppsci/loss/mtl/__init__.py:30]: 已删除cfg的typehint，跟其他build函数保持一致，给每一个build函数import一个AttrDict感觉不太有必要……

**Review by zhiminzhang0830 — APPROVED** [review]
LGTM
